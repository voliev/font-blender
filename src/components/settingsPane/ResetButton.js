import React from "react";
import PropTypes from "prop-types";
import "./styles/ResetButton.css";

const ResetButton = ({ disabled, resetButtonClickHandler }) => (
  <button
    className="ResetButton btn-no-background"
    disabled={disabled}
    onClick={resetButtonClickHandler}
    title="Reset styles"
    value="reset"
  >
    Reset
  </button>
);

ResetButton.propTypes = {
  disabled: PropTypes.bool.isRequired,
  resetButtonClickHandler: PropTypes.func.isRequired
};

export default ResetButton;
