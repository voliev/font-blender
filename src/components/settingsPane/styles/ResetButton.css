.ResetButton {
  margin: 0 0 0 var(--space-md);
  padding: var(--space-xs) var(--space-sm);
}
