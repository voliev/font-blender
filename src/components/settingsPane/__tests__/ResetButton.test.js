import React from "react";
import { mount, shallow } from "enzyme";
import ResetButton from "../ResetButton";

const setup = (renderFn = shallow, propsOverrides = {}) => {
  const props = {
    disabled: false,
    resetButtonClickHandler: jest.fn(),
    ...propsOverrides
  };

  return {
    ...props,
    wrapper: renderFn(<ResetButton {...props} />)
  };
};

afterEach(() => {
  jest.clearAllMocks();
});

it("should render without crashing", () => {
  const { wrapper } = setup();

  expect(wrapper).toMatchSnapshot();
});

it("should receive corresponding props", () => {
  const { disabled, resetButtonClickHandler, wrapper } = setup(mount);

  expect(typeof wrapper.prop("disabled")).toBe("boolean");
  expect(wrapper.prop("disabled")).toBe(disabled);

  expect(typeof wrapper.prop("resetButtonClickHandler")).toBe("function");
  expect(wrapper.prop("resetButtonClickHandler")).toBe(resetButtonClickHandler);
});

it("should handle button click", () => {
  const { resetButtonClickHandler, wrapper } = setup();

  wrapper.find("button").simulate("click");

  expect(resetButtonClickHandler).toHaveBeenCalledTimes(1);
});
