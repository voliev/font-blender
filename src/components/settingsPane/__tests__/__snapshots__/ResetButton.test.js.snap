// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render without crashing 1`] = `
<button
  className="ResetButton btn-no-background"
  disabled={false}
  onClick={[MockFunction]}
  title="Reset styles"
  value="reset"
>
  Reset
</button>
`;
